### PR TITLE
Z registration

### PR DESCRIPTION
**Pull request by Zahra120**

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "development": {
+    "username": "zahra120",
```
Comment (izelnakri) [inline config/config.json]: get this from process.ENV.DB_USERNAME because william has a different username in his own local computer.

Comment (izelnakri) [reply config/config.json]: you need to turn the extension from .json to .js to achive this

File: index.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+app.set('view engine', 'pug');
+
+ app.get('/', (req, res) => {
+    res.render('users/new');
+ });
+app.get('/login', (req, res) => {
+  res.redirect('/admin');
+});
+
+app.get('/admin', (req, res) => {
```
Comment (izelnakri) [inline index.js]: You probably dont need an admin panel for this application. Do you need one?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  res.render('users/login');
+});
+app.get('/challenge', (req, res) => {
+  res.render('challenge/index');
+});
+app.get('/admin/challenge/new', (req, res) => {
```
Comment (izelnakri) [inline index.js]: Same here, you probably dont need an admin panel for this application.

File: index.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+app.post('/users', (req, res) => {
+  var user = req.body;
+  db.User.create(user).then((user) => {
+      res.redirect('/admin');
+  });
+});
+app.post('/login', (req, res) => {
+
+});
+app.post('/challenge', (req, res) => {
```
Comment (izelnakri) [inline index.js]: POST /challenges

File: views/challenge/index.pug [hunk 5]
```diff
@@ -0,0 +1,14 @@
+extends ../application.pug
+block content
+	.container
+		.row
+			.col-xs-8.offset-xs-4
+				if(challenge)
```
Comment (izelnakri) [inline views/challenge/index.pug]: remove the paranthesis

File: views/challenge/new.pug [hunk 6]
```diff
@@ -0,0 +1,23 @@
+extends ../application.pug
+block content
+	.container
+		.row
+			.col-xs-8.offset-xs-2
+				form(action= "/challenge" method= "POST")
+					.form-group
+						label(for="")  Name of challenge
```
Comment (izelnakri) [inline views/challenge/new.pug]: put the right for="" to the input ids. and then put the right input types

**Comment by izelnakri** [comment]
Do these changes Zahra then, when the registration feature is complete we can merge this to master.

File: index.js (hunk 2 above)
Comment (Zahra120) [reply index.js]: Only users should be able to see the challenges and create one. so for that, we need the admin, right?

Comment (izelnakri) [reply index.js]: no you dont, you can handle that logic in the route handler, or in a middleware

File: index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
+const express = require('express'),
+      pug     = require('pug'),
+      morgan = require('morgan'),
+      sequelize = require('sequelize'),
+      nodemailer = require('nodemailer'),
+      bodyParser = require('body-parser'),
+      displayRoutes = require('express-routemap'),
+       pg = require('pg'),
```
Comment (izelnakri) [inline index.js]: work on your indention

File: index.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+app.use(morgan('dev'));
+app.use(session({
+  secret: 'keyboard cat',
+  resave: true,
+  saveUninitialized: true,
+  cookie: { secure: true }
+}));
+app.set('view engine', 'pug');
+
+ app.get('/', (req, res) => {
```
Comment (izelnakri) [inline index.js]: work on your indentation

File: index.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+//
+
+//   });
+});
+db.sequelize.sync().then(() => {
+  app.listen(3000, () => {
+      console.log('Web Server is running on port 3000');
+       displayRoutes(app);
+    });
+  });
```
Comment (izelnakri) [inline index.js:85]: work on your indentation

File: migrations/20161127171551-create-user.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      name: {
+        type: Sequelize.STRING
+      },
+      surname: {
+        type: Sequelize.STRING
+      },
+      email: {
+        type: Sequelize.STRING
+      },
+      password: {
```
Comment (izelnakri) [inline migrations/20161127171551-create-user.js]: this should be named passwordDigest, also in the model

File: migrations/20161129093642-create-challenge.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+      active: {
+        type: Sequelize.BOOLEAN
+      },
+      pot: {
+        type: Sequelize.INTEGER
+      },
+      length: {
+        type: Sequelize.INTEGER
+      },
+      winnerId: {
```
Comment (izelnakri) [inline migrations/20161129093642-create-challenge.js]: Sequelize convention is to use RelationshipId for foreign keys, thus: WinnerId.

File: migrations/20161129103720-change-name-length.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+    queryInterface.renameColumn('Challenges', 'length', 'numberOfDay');
```
Comment (izelnakri) [inline migrations/20161129103720-change-name-length.js:5]: numberOfDays not numberOfDay

File: migrations/20161130140608-create-foreignkey-challenge-in-Task.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+    queryInterface.addColumn('Tasks','ChallengeId',Sequelize.INTEGER);
```
Comment (izelnakri) [inline migrations/20161130140608-create-foreignkey-challenge-in-Task.js]: put spaces after your commas

File: migrations/20161130140932-create-foreignkey-User-in-Task.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+      queryInterface.addColumn('Tasks','UserId',Sequelize.INTEGER);
```
Comment (izelnakri) [inline migrations/20161130140932-create-foreignkey-User-in-Task.js]: put spaces after your commas

File: models/challenge.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+module.exports = function(sequelize, DataTypes) {
+  var Challenge = sequelize.define('Challenge', {
+    name: DataTypes.STRING,
+    active: DataTypes.BOOLEAN,
+    pot: DataTypes.INTEGER,
+    numberOfDay: DataTypes.INTEGER,
```
Comment (izelnakri) [inline models/challenge.js]: numberOfDays

File: models/challenge.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+    winnerId: DataTypes.INTEGER
```
Comment (izelnakri) [inline models/challenge.js]: you dont need relationship references here

File: views/users/login.pug [hunk 17]
```diff
@@ -0,0 +1,14 @@
+extends ../application.pug
+block content
+	.container
+		.row
+			.col-xs-8.offset-xs-2
+				form(action= "/login" method= "POST")
+					.form-group
+						label(for="")  Email
+						input(type="" class="form-control" name="email" )
```
Comment (izelnakri) [inline views/users/login.pug]: put input types and label for="" id references along with input ids

File: views/users/new.pug [hunk 18]
```diff
@@ -0,0 +1,20 @@
+extends ../application.pug
+block content
+	.container
+		.row
+			.col-xs-8.offset-xs-2
+				form(action= "/users" method= "POST")
+					.form-group
+						label(for="")  Name
```
Comment (izelnakri) [inline views/users/new.pug]: put input types and label for="" id references along with input ids

File: index.js [hunk 19]
```diff
@@ -0,0 +1,110 @@
+const express = require('express'),
+      pug     = require('pug'),
+      morgan = require('morgan'),
+      sequelize = require('sequelize'),
```
Comment (izelnakri) [inline index.js]: you dont need this package here, so remove it.

File: index.js [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+  cookie: { secure: true }
+}));
+app.set('view engine', 'pug');
+
+app.get('/', (req, res) => {
+    res.render('users/new');
+
+ });
+
+app.get('/login', (req, res) => {
```
Comment (izelnakri) [inline index.js]: move login/ logout routes to a new route file.

File: index.js [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+  db.Challenge.create(req.body.challenge).then((challenge) => {
+    db.Task.create({
+      name: req.body.task.name,
+      userId: 1, // change this to req.session.user.id for the right logic
+      challengeId: challenge.id
+    }).then((task) => {
+      db.User.create({
+        email: req.body.participant.email
+      }).then((participant) => {
+        console.log(participant);
```
Comment (izelnakri) [inline index.js:60]: you can send an email here but to this at the very last, focus on the login/register functionality for now.

File: views/users/login.pug [hunk 22]
```diff
@@ -0,0 +1,14 @@
+extends ../application.pug
+block content
+	.container
+		.row
+			.col-xs-8.offset-xs-2
+				form(action= "/login" method= "POST")
+					.form-group
+						label(for="email")  Email
+						input(type="email" class="form-control" id ="email" name="email" )
```
Comment (izelnakri) [inline views/users/login.pug]: remove the space in the id here

**Pull request merged**